client/app/account/billing/confirmTerminate/billing-confirmTerminate.controller.js

```diff
@@ -1,57 +1,81 @@
+            "MIGRATED_TO_COMPETITOR",
+            "MIGRATED_TO_ANOTHER_OVH_PRODUCT",
+            "OTHER"
+        ];
 
-    this.serviceId = $stateParams.id;
-    this.serviceState = null;
-    this.token = $stateParams.token;
-    this.loading = true;
-    this.terminating = false;
-    this.error = false;
-    this.globalError = null;
+        this.serviceId = $stateParams.id;
+        this.serviceState = null;
+        this.token = $stateParams.token;
+        this.loading = true;
+        this.userLoading = true;
+        this.terminating = false;
+        this.error = false;
+        this.globalError = null;
 
-    this.init = function () {
-        if (!this.token || !this.serviceId) {
-            this.globalError = true;
-            return;
-        }
+        this.init = function () {
+            if (!this.token || !this.serviceId) {
+                this.globalError = true;
+                return;
+            }
 
-        this.loading = true;
-        BillingTerminate
-            .getServiceInfo(this.serviceId)
-            .then((serviceInfos) => {
-                this.serviceInfos = serviceInfos;
-                return serviceInfos.serviceId;
-            })
-            .then((serviceId) =>
-                BillingTerminate.getServiceApi(serviceId, true).then((service) => {
-                    this.serviceState = _.get(service, "resource.state");
-                })
-            )
-            .catch(() => {
+            $q.all([
+                this.loadUser(),
+                this.loadService()
+            ]).catch(() => {
                 this.globalError = true;
-            })
-            .finally(() => {
-                this.loading = false;
-            });
-    };
-
-    this.confirmTermination = function () {
-        this.terminating = true;
-        BillingTerminate
-            .confirmTermination(this.serviceId, this.serviceInfos.domain, this.reason, this.commentary, this.token)
-            .then(() => {
-                this.error = false;
-                this.serviceState = "suspending";
-            })
-            .catch(() => {
-                this.error = true;
-            })
-            .finally(() => {
-                this.terminating = false;
             });
-    };
+        };
+
+        this.loadUser = function () {
+            return User.getUser()
+                .then((user) => {
+                    this.USVersion = user && user.ovhSubsidiary === "US";
+                })
+                .finally(() => {
+                    this.userLoading = false;
+                });
+        };
+
+        this.loadService = function () {
+            return BillingTerminate
+                .getServiceInfo(this.serviceId)
+                .then((serviceInfos) => {
+                    this.serviceInfos = serviceInfos;
+                    return serviceInfos.serviceId;
+                })
+                .then((serviceId) => BillingTerminate.getServiceApi(serviceId, true).then((service) => {
```

Comment:
The second `then` should be chained instead

Reply:
Yep can be done.